libnestutil/propagator_stability.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ *  propagator_stability.cpp
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "propagator_stability.h"
+#include <cmath>
+
+double
+propagator_32( double tau_syn, double tau, double C, double h )
+{
+  const double P32_linear =
+    1 / ( 2. * C * tau * tau ) * h * h * ( tau_syn - tau ) * std::exp( -h / tau );
+  const double P32_singular = h / C * std::exp( -h / tau );
+  const double P32 =
+    1 / C * ( std::exp( -h / tau ) - std::exp( -h / tau_syn ) ) / ( -1 / tau - -1 / tau_syn );
+  const double dev_P32 = std::abs( P32 - P32_singular );
+
+  if ( tau == tau_syn )
+  {
+    return P32_singular;
+  }
+  else if ( std::abs( tau - tau_syn ) < 0.1 and dev_P32 > 2 * P32_linear )
```

Comment:
The if-else-if here is unnecessary and, in my opinion, confusing rather than clarifying. You should combine the condition for using the singular value in a Boolean or.